doc/pages/architecture/080-mqtt-v1-protocol.md

```diff
@@ -0,0 +1,278 @@
+| double            | double (0x01)       | 8                                                  |
+| integer           | int32 (0x10)        | 4                                                  |
+| boolean           | boolean (0x08)      | 1                                                  |
+| longinteger       | int64 (0x12)        | 8                                                  |
+| string            | UTF-8 string (0x02) | >= length (encoding dependent)                     |
+| binaryblob        | binary (0x05)       | length                                             |
+| datetime          | UTC datetime (0x09) | 8                                                  |
+| doublearray       | Array (0x04)        | (8 + keysize) * count                              |
+| integerarray      | Array (0x04)        | (4 + keysize) * count                              |
+| booleanarray      | Array (0x04)        | (1 + keysize) * count                              |
+| longintegerarray  | Array (0x04)        | (1 + keysize) * count                              |
+| stringarray       | Array (0x04)        | depends on count, length, keys length and encoding |
+| binaryblobarray   | Array (0x4)         | depends on count, keys length and length           |
+
+Object aggregations are encoded as embedded documents.
+
+## Connection and Disconnection
+
+A device is not required to publish any additional connection or disconnection message, the MQTT broker automatically keeps track of those events.
+When connecting, before publishing any data message, a device should check MQTT _session present_ flag. When _session present_ is _true_ no further actions are required, when _false_ the device should take following actions:
+
+* Publish introspection
+* Publish empty cache
+* Publish all set properties on all property interfaces
+
+If a device is unable to inspect _session present_ all previous actions must be taken at every reconnection.
+
+## Introspection
+
+Each device must declare the set of supported interfaces and their version. Astarte needs to know the interfaces before processing any further data publish.
+This message in Astarte jargon is called _introspection_ and it's performed by publishing on the device root topic the list of supported interfaces.
+
+Introspection payload is a simple plain text string, and it has the following format (in BNF like syntax):
+
+```
+introspection ::= introspection_list
+introspection_list ::= introspection_entry ";" introspection_list | introspection_entry
+introspection_entry ::= interface_name ":" interface_major_version ":" interface_minor_version
+```
+
+The following example is a valid introspection payload:
+
+```
+com.example.MyInterface:1:0;org.example.DraftInterface:0:3
+```
+
+## Empty Cache
+
+A device may declare to have no knowledge about remote set properties. This usually happens on a new device, after a factory reset or when cache is lost.
```

Comment:
I'd rephrase.

Astarte MQTT v1 strives to save bandwidth upon reconnections, to make sure even frequent reconnections don't affect bandwidth consumption. As such, upon connecting and if MQTT advertises a session present, both sides assume that data flow is ordered and consistent. However, there might be cases where this guarantee isn't respected by the device for a number of reasons (e.g.: new device, factory reset, cache lost...). In this case, a device might declare that it has no confidence about its status and its known properties, and can request to resynchronise entirely with Astarte.